core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.spec.ts

```diff
@@ -247,6 +249,50 @@ describe('Translation Modal Component', () => {
     });
   });
 
+  describe('with a machine translatable translation language', () => {
+    beforeEach(fakeAsync(() => {
+      translationLanguageService.setActiveLanguageCode('es');
+      component.ngOnInit();
+      const sampleStateWiseContentMapping = {
+        stateName1: {contentId1: 'Please continue'}
+      };
+
+      const req = httpTestingController.expectOne(
+        '/gettranslatabletexthandler?exp_id=1&language_code=es');
+      expect(req.request.method).toEqual('GET');
+      req.flush({
+        state_names_to_content_id_mapping: sampleStateWiseContentMapping,
+        version: 1
+      });
+      flushMicrotasks();
+    }));
+
+    it('should offer translation', () => {
+      expect(component.languageSupportsMachineTranslations).toBeTrue();
+    });
+
+    it('should not retrieve translation until dropdown chevron is clicked',
+      fakeAsync(() => {
+        expect(component.hideMachineTranslation).toBeTrue();
+        component.toggleMachineTranslation();
+        const sampleMachineTranslationResponse:
+        MachineTranslatedTextBackendDict = {
+          translated_texts: {contentId1: new TranslatableItem(
+            'Por favor continua.', 'html', 'content', null, null)}
+        };
+        const req = httpTestingController.expectOne(
+          '/machine_translated_state_texts_handler?exp_id=1&state_name=' +
+          'stateName1&content_ids=%5B%22contentId1%22%5D&' +
+          'target_language_code=es');
```

Comment:
Can we add new lines before this and after line 289 to delineate the test phases (arrange, act, assert)?